pages/api/friends.js

```diff
@@ -0,0 +1,36 @@
+const { Friend, User } = require('../../models');
+
+export default async (req, res) => {
+  if (req.method === 'POST') {
+    let user_id = req.body.user_id;
+    let user = await User.findByPk(user_id);
+
+    if (user) {
+      let existingFriend = await Friend.findAll({
+        where: { user_id: user.id, name: req.body.name },
+      });
+
+      if (
+        existingFriend &&
+        existingFriend.length &&
+        existingFriend[0].name === req.body.name
+      ) {
+        return res.status(409).json({
+          httpCode: 409,
+          message: 'Friend with this name already exists',
+        });
+      } else {
+        res.json(await Friend.create(req.body));
```

Comment:
Can you allow bulk insertion of more than 1 friend at a time? I think you can use `Friend.bulkCreate` and pass it an array. https://sequelize.org/master/manual/model-querying-basics.html#creating-in-bulk

Reply:
I'm working on it in a separate branch

Reply:
kk